### DISCUSSION TITLE
multiSelect INPUT and numbers as options kind of breaks the bound Property ? 🤔

### DISCUSSION THREAD

**Discussion by I-Pch** [mProjectsCode/obsidian-meta-bind-plugin#300] [Q&A] [answered]
Hi there 😄 !

I wasn't sure where to ask and as I'm brand new Meta Bind user maybe I just did something not totally right, leading to some unexpected results 😇 ... 

And my question is:
 
Is it normal/expected that using numbers for `multiSelect` options in `Meta Bind` kind of break the bound properties ? 

<img width="1202" alt="MB - Numbers - Multi" src="https://github.com/mProjectsCode/obsidian-meta-bind-plugin/assets/77828085/dda6c077-3933-49ff-9f9d-6fcddc450291">

No matter how I write the `options` in the input field, I don't seem to be able to find a way to keep the "Obsidian formatting/display" ... 
I mean, when manually entering a number in a `list` type of key from Properties, in Source, Obsidian stores it as a string (`"1"`) but displays it as it was input (`1`) ... 

But, none of these 3 `INPUT` seem to just add `"1"` as a value in Source... :thinking: 
 
This one seem to just output numbers ... which breaks the property: 
```javascript
~~~meta-bind
INPUT[multiSelect(option(0), option(1)):multi]
~~~
```
(Result in Source)
```yaml
multi:
  - 0
  - 1
```

This one still output numbers despite the single quotes   
```javascript
~~~meta-bind
INPUT[multiSelect(option('0'), option('1')):multi2]
~~~
```
(Result in Source)
```yaml
multi2:
  - 0
  - 1
```

And this one, surrounds the options by single quotes leading Properties to display the double ones (`"0"`) 😊 ... 

```javascript
~~~meta-bind
INPUT[multiSelect(option("0"), option("1")):multi3]
~~~
```
(Result in Source)
```yaml
multi3:
  - '"0"'
  - '"1"'
```

Isn't there a way to use numeric values as `options` in a `multiSelect` and have them being written, in Source, as: 
```yaml
manualMulti:
  - "1"
  - "2"
  - "3"
  - "4"
```  
... ? 🤔 

I saw that because I was trying to create a multi select input where the options are the folders I have in my vault and one of these was named `000` 😅 ... Selecting it lead to the same results: it "somewhat" broke the key 😇 (I say "somewhat" because when I unselect `000`, everything's back to normal)...

https://github.com/mProjectsCode/obsidian-meta-bind-plugin/assets/77828085/9172d5b7-d378-4ec7-b8ab-028619789529

**Comment by mProjectsCode** [comment, marked as answer]
Yeah, this is somewhat expected. MB interprets numbers as numbers and `true`/`false` as booleans. Single quoted strings only exist to allow otherwise forbidden characters in the argument value. I would have expected the properties UI to simply work with lists of numbers... 😅

**Reply by I-Pch** [reply]
Thank you for confirming that I didn't do anything wrong when it comes to Meta Bind 😁 ! 

> I would have expected the properties UI to simply work with lists of numbers... 😅

I would have thought too when Properties became available 😄... Maybe there's a "YAML specifications" reason behind this 🤷‍♀️ 😄.

On the other hand this is not an issue in the sense that I don't normally have folders titled as `000` 😊 ... I just named that one like  this because I got lazy when creating & naming "tests" folders ... 
Now I know that I just need to add at least one letter to the name of a folder 😄
